src/util/sandbox/fd_sandbox.h

```diff
@@ -0,0 +1,30 @@
+#ifdef FD_HAS_SANDBOX
+
+#ifndef HEADER_fd_src_util_sandbox_fd_sandbox_h
+#define HEADER_fd_src_util_sandbox_fd_sandbox_h
+
+#include "../fd_util_base.h"
+
+#include <linux/filter.h>
+#include <stdbool.h>
+
+struct fd_sandbox_profile {
+  bool                initialized;
+
+  char *              chroot_path;
+  uint                close_fds_beyond;
+  uint                max_open_fds;
+  char *              netns;
+  struct sock_fprog * seccomp_prog;
+  char *              user;
+};
+typedef struct fd_sandbox_profile fd_sandbox_profile_t;
+
+/* fd_sandbox sandboxes the current process. */
+int fd_sandbox              ( const fd_sandbox_profile_t * const sandbox_profile );
+
+/* fd_sandbox_profile_init initializes a fd_sandbox_profile_t* with sane defaults. */
+void fd_sandbox_profile_init ( fd_sandbox_profile_t * profile );
```

Comment:
Getting further along here, feels like there wants to be a sandbox profile that just exists by default for each application thread group (e.g. Linux PID).  On targets without FD_HAS_SANDBOX, this profile would be initialized during fd_boot (after the log has gone live) to indicate to the thread group there is no sandboxing.

Similarly, on targets with FD_HAS_SANDBOX, this profile would be set to sane defaults if no parameters were specified (e.g. no sandboxing) and admin defaults parameters passed by the command line (stripped) / environment. otherwise.

Then then would be some additional APIs that allow sandboxing aware code to further restrict their permissions after they do initialization.  (E.g. setup all shared memory pages and then drop various syscall permissions so that running code can't change any internal memory region firewalling.)

(The boot and and app requested changes should be extensively logged as well to aid in debugging sandboxing, understanding fine grained permissions, etc.)